actiontext/lib/templates/installer.rb

```diff
@@ -19,7 +21,14 @@
   "app/views/active_storage/blobs/_blob.html.erb"
 
 say "Installing JavaScript dependencies"
-run "yarn add #{JS_DEPENDENCIES.map { |name, version| "#{name}@#{version}" }.join(" ")}"
+
+begin
+  exec "yarnpkg add #{JS_DEPENDENCIES.map { |name, version| "#{name}@#{version}" }.join(" ")}"
```

Comment:
We have the same check in `bin/yarn`. Can we reuse it?  

Also, I think that if `yarn` not exists, should not do other operations too.  
How do you think that run `yarn` first and if it fails do not run other operations? 

I am thinking about the following.

```diff 
diff --git a/actiontext/lib/tasks/actiontext.rake b/actiontext/lib/tasks/actiontext.rake
index 4f90e49..b8fb81f 100644
--- a/actiontext/lib/tasks/actiontext.rake
+++ b/actiontext/lib/tasks/actiontext.rake
@@ -10,6 +10,7 @@ namespace :action_text do
   task :run_installer do
     installer_template = File.expand_path("../templates/installer.rb", __dir__)
     system "#{RbConfig.ruby} ./bin/rails app:template LOCATION=#{installer_template}"
+    abort unless $?.success?
   end
 
   task :copy_migrations do
diff --git a/actiontext/lib/templates/installer.rb b/actiontext/lib/templates/installer.rb
index a8000eb..3dae8bb 100644
--- a/actiontext/lib/templates/installer.rb
+++ b/actiontext/lib/templates/installer.rb
@@ -8,6 +8,9 @@
 JS_DEPENDENCIES = JS_PACKAGE["peerDependencies"].dup.merge \
   JS_PACKAGE["name"] => "^#{JS_PACKAGE["version"]}"
 
+say "Installing JavaScript dependencies"
+run "bin/yarn add #{JS_DEPENDENCIES.map { |name, version| "#{name}@#{version}" }.join(" ")}", abort_on_failure: true
+
 say "Copying actiontext.scss to app/assets/stylesheets"
 copy_file "#{__dir__}/actiontext.scss", "app/assets/stylesheets/actiontext.scss"
 
@@ -18,9 +21,6 @@
 copy_file "#{__dir__}/../../app/views/active_storage/blobs/_blob.html.erb",
   "app/views/active_storage/blobs/_blob.html.erb"
 
-say "Installing JavaScript dependencies"
-run "yarn add #{JS_DEPENDENCIES.map { |name, version| "#{name}@#{version}" }.join(" ")}" 
```

Reply:
@y-yagi Yeah, it would be best to reuse the code (I got the snippet from bin/yarn). My line of thought for not using bin/yarn was: would an existing application that is upgrading have the bin/yarn file?

If you don't think that's an issue for existing applications, then certainly using bin/yarn is the best approach. Let me know your thoughts and I will patch the implementation.

Reply:
I’m pretty sure app:update generates bin/yarn, but there could be an order issue.

Reply:
The `app:update` command generates `bin/yarn` definitely. And `bin/yarn` introduced in Rails 5.1. There should be a chance to upgrade twice, so I guess it probably does not matter.  

However, it may be advisable to first check that there is `bin/yarn`, then run `app:update` if it does not exist. I think this will be more user-friendly.

Reply:
Don’t think automatically running app:update is a good idea, since there’s usually lots of decisions you have to make when running it.

We could abort and say users should run app:update. But maybe that’s not worth it either.